titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/service/action/BasicServiceJobActions.java

```diff
@@ -39,19 +46,47 @@
     /**
      * Update job capacity.
      */
-    public static TitusChangeAction updateJobCapacityAction(ReconciliationEngine<JobManagerReconcilerEvent> engine, Capacity capacity, JobStore jobStore, CallMetadata callMetadata) {
+    public static TitusChangeAction updateJobCapacityAction(ReconciliationEngine<JobManagerReconcilerEvent> engine,
+                                                            CapacityAttributes capacityAttributes,
+                                                            JobStore jobStore,
+                                                            CallMetadata callMetadata,
+                                                            EntitySanitizer entitySanitizer) {
         return TitusChangeAction.newAction("updateJobCapacityAction")
                 .id(engine.getReferenceView().getId())
                 .trigger(V3JobOperations.Trigger.API)
-                .summary("Changing job capacity to: %s", capacity)
+                .summary("Changing job capacity to: %s", capacityAttributes)
                 .changeWithModelUpdates(self -> {
                     Job<ServiceJobExt> serviceJob = engine.getReferenceView().getEntity();
+
                     if (serviceJob.getStatus().getState() != JobState.Accepted) {
                         return Observable.error(JobManagerException.jobTerminating(serviceJob));
                     }
 
-                    Job<ServiceJobExt> updatedJob = JobFunctions.changeServiceJobCapacity(serviceJob, capacity);
+                    Capacity.Builder newCapacityBuilder = serviceJob.getJobDescriptor().getExtensions().getCapacity().toBuilder();
+                    Evaluators.acceptIfTrue(capacityAttributes.getDesired().isPresent(), valueAccepted -> newCapacityBuilder.withDesired(capacityAttributes.getDesired().get()));
```

Comment:
same comment as before: this can be simplified with `Optional#ifPresent(consumer)`. Since it's being used more than once, it may be worth extracting a common builder/converter to avoid the code duplication